flashinfer/gemm.py

```diff
@@ -500,8 +547,13 @@ def fp8_gemm_sm100(
 
 
 @functools.cache
-def get_gemm_sm100_module_cutlass_fp4():
-    module = gen_gemm_sm100_module_cutlass_fp4().build_and_load()
+def get_gemm_module_cutlass_fp4():
+    # Check if we're on SM120/121 and use the appropriate module
+    major, _ = get_compute_capability(torch.device("cuda"))
```

Comment:
I don't encourage doing this because different device might have different architectures.

You can separate two functions:
- get_gemm_sm100_module_cutlass_fp4
- get_gemm_sm120_module_cutlass_fp4

and dispatch to different functions in the caller side by determining the device of input tensor.